src/neo/IO/ByteArrayWrapper.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Neo.IO
+{
+    public class ByteArrayWrapper : ICloneable<ByteArrayWrapper>, IEquatable<ByteArrayWrapper>, IEquatable<byte[]>, ISerializable
+    {
+        public byte[] Value;
+
+        public ByteArrayWrapper()
+        {
+        }
+
+        public ByteArrayWrapper(byte[] value)
+        {
+            this.Value = value;
+        }
+
+        public int Size => Value is null ? 0 : Value.Length;
+
+        public ByteArrayWrapper Clone()
+        {
+            return new ByteArrayWrapper(Value);
+        }
+
+        public void FromReplica(ByteArrayWrapper replia)
+        {
+            this.Value = replia.Value;
+        }
+
+        public unsafe void Deserialize(BinaryReader reader)
+        {
+            Value = reader.ReadVarBytes();
+        }
+
+        public bool Equals(byte[] other)
+        {
+            if (other is null || Value is null) return false;
+            return Value.SequenceEqual<byte>(other);
+        }
+
+        public bool Equals(ByteArrayWrapper other)
+        {
+            if (other is null || other.Value is null || Value is null) return false;
```

Comment:
If `Value` is null and `other.Value` is null should be `true`

Reply:
done.